From e3b0522b6e51351939813168dc73866f59af6a0f Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Mon, 10 May 2021 16:04:05 +0300
Subject: [PATCH 07/11] d4xx: Add support of Y8 and Y12I formats

 - Y8 and Y12I are exposed on the same node of Z16
 - Remove res 2000x1500 from YUYV format

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c                      | 101 +++++++-----------
 .../platform/tegra/camera/vi/vi4_formats.h    |   4 +-
 2 files changed, 39 insertions(+), 66 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 9ec09b023..11dbffb56 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -484,11 +484,6 @@ static const struct ds5_resolution ds5_rgb_sizes[] = {
 		.height = 1080,
 		.framerates = ds5_framerates,
 		.n_framerates = ARRAY_SIZE(ds5_framerates),
-	},  {
-		.width = 2000,
-		.height = 1500,
-		.framerates = ds5_framerates,
-		.n_framerates = ARRAY_SIZE(ds5_framerates),
 	},
 };
 
@@ -505,12 +500,6 @@ static const struct ds5_resolution d43x_calibration_sizes[] = {
 		.height = 800,
 		.framerates = ds5_framerate_15_30,
 		.n_framerates = ARRAY_SIZE(ds5_framerate_15_30),
-	}, {
-		.width =  640,
-		.height = 800,
-		.framerates = ds5_framerate_15_30,
-		.n_framerates = ARRAY_SIZE(ds5_framerate_15_30),
-
 	},
 };
 
@@ -523,13 +512,6 @@ static const struct ds5_resolution d46x_calibration_sizes[] = {
 	},
 };
 
-static struct ds5_resolution ds5_size_calibration_wide = {
-	.width =  1280,
-	.height = 800,
-	.framerates = &ds5_framerate_25,
-	.n_framerates = 1,
-};
-
 static const struct ds5_format ds5_depth_formats_d43x[] = {
 	{
 		// TODO: 0x31 is replaced with 0x1e since it caused low FPS in Jetson.
@@ -537,7 +519,17 @@ static const struct ds5_format ds5_depth_formats_d43x[] = {
 		.mbus_code = MEDIA_BUS_FMT_UYVY8_1X16,
 		.n_resolutions = ARRAY_SIZE(d43x_depth_sizes),
 		.resolutions = d43x_depth_sizes,
-	},
+	}, {
+		.data_type = 0x2a,	/* Y8 */
+		.mbus_code = MEDIA_BUS_FMT_Y8_1X8,
+		.n_resolutions = ARRAY_SIZE(d43x_depth_sizes),
+		.resolutions = d43x_depth_sizes,
+	}, {
+		.data_type = 0x24,	/* 24-bit Calibration */
+		.mbus_code = MEDIA_BUS_FMT_RGB888_1X24,	/* FIXME */
+		.n_resolutions = ARRAY_SIZE(d43x_calibration_sizes),
+		.resolutions = d43x_calibration_sizes,
+    },
 };
 
 static const struct ds5_format ds5_depth_formats_d46x[] = {
@@ -547,50 +539,33 @@ static const struct ds5_format ds5_depth_formats_d46x[] = {
 		.mbus_code = MEDIA_BUS_FMT_UYVY8_1X16,
 		.n_resolutions = ARRAY_SIZE(d46x_depth_sizes),
 		.resolutions = d46x_depth_sizes,
-	},
+	}, {
+		/* First format: default */
+		.data_type = 0x2a,	/* Y8 */
+		.mbus_code = MEDIA_BUS_FMT_Y8_1X8,
+		.n_resolutions = ARRAY_SIZE(d46x_depth_sizes),
+		.resolutions = d46x_depth_sizes,
+	}, {
+		.data_type = 0x24,	/* 24-bit Calibration */
+		.mbus_code = MEDIA_BUS_FMT_RGB888_1X24,	/* FIXME */
+		.n_resolutions = ARRAY_SIZE(d46x_calibration_sizes),
+		.resolutions = d46x_calibration_sizes,
+    },
 };
 
 #define DS5_DEPTH_N_FORMATS 1
 
 static const struct ds5_format ds5_y_formats_ds5u[] = {
-		{
-				.data_type = 0x1e,	/* UYVY */
-				.mbus_code = MEDIA_BUS_FMT_UYVY8_2X8,
-				.n_resolutions = ARRAY_SIZE(ds5_sizes),
-				.resolutions = ds5_sizes,
-			},
-};
-
-static struct ds5_format ds5_y_formats_asr[] = {
 	{
-		.data_type = 0x1e,	/* UYVY */
-		.mbus_code = MEDIA_BUS_FMT_UYVY8_2X8,
-		.n_resolutions = ARRAY_SIZE(ds5_sizes),
-		.resolutions = ds5_sizes,
-	},
-};
-
-static const struct ds5_format ds5_y_formats_awg[] = {
-	{
-		.data_type = 0x1e,	/* UYVY */
-		.mbus_code = MEDIA_BUS_FMT_UYVY8_2X8,
-		.n_resolutions = ARRAY_SIZE(ds5_sizes),
-		.resolutions = ds5_sizes,
-	}, {
 		.data_type = 0x2a,	/* Y8 */
 		.mbus_code = MEDIA_BUS_FMT_Y8_1X8,
 		.n_resolutions = ARRAY_SIZE(ds5_sizes),
 		.resolutions = ds5_sizes,
 	}, {
-		.data_type = 0x30,	/* 24-bit Calibration */
-		.mbus_code = MEDIA_BUS_FMT_FIXED,	/* FIXME */
-		.n_resolutions = 1,
-		.resolutions = &ds5_size_calibration_wide,
-	}, {
-		.data_type = 0x32,	/* 16-bit RY8_LY8 Calibration */
-		.mbus_code = MEDIA_BUS_FMT_UYVY8_1X16,
-		.n_resolutions = ARRAY_SIZE(ds5_sizes),
-		.resolutions = ds5_sizes,
+        .data_type = 0x24,	/* 24-bit Calibration */
+		.mbus_code = MEDIA_BUS_FMT_RGB888_1X24,	/* FIXME */
+		.n_resolutions = ARRAY_SIZE(d46x_calibration_sizes),
+		.resolutions = d46x_calibration_sizes,
 	},
 };
 
@@ -607,14 +582,6 @@ static const struct ds5_variant ds5_variants[] = {
 		.formats = ds5_y_formats_ds5u,
 		.n_formats = ARRAY_SIZE(ds5_y_formats_ds5u),
 	},
-	[DS5_ASR] = {
-		.formats = ds5_y_formats_asr,
-		.n_formats = ARRAY_SIZE(ds5_y_formats_asr),
-	},
-	[DS5_AWG] = {
-		.formats = ds5_y_formats_awg,
-		.n_formats = ARRAY_SIZE(ds5_y_formats_awg),
-	},
 };
 
 static const struct v4l2_mbus_framefmt ds5_mbus_framefmt_template = {
@@ -884,9 +851,15 @@ static int ds5_configure(struct ds5 *state)
 		//       this will configure FW to override DT in mipi header to
 		//       0x1E for depth, since DT 0x31 is not supported
 		//       in Jetson
-		ret = ds5_write(state, 0x401C, 0x1E);
-		if (ret < 0)
-			return ret;
+        if (dfmt) {
+			ret = ds5_write(state, 0x401C, 0x1E);
+			if (ret < 0)
+				return ret;
+		} else {
+			ret = ds5_write(state, 0x401C, mfmt);
+			if (ret < 0)
+				return ret;
+		}
 
 		ret = ds5_write(state, DS5_DEPTH_Y_FPS, depth->config.framerate);
 		if (ret < 0)
@@ -2530,7 +2503,7 @@ static int ds5_fixed_configuration(struct i2c_client *client, struct ds5 *state)
 	default:
 		sensor->formats = ds5_depth_formats_d46x;
 	}
-	sensor->n_formats = DS5_DEPTH_N_FORMATS;
+	sensor->n_formats = 3;
 	sensor->mux_pad = DS5_MUX_PAD_DEPTH;
 
 	sensor = &state->motion_t.sensor;
diff --git a/drivers/media/platform/tegra/camera/vi/vi4_formats.h b/drivers/media/platform/tegra/camera/vi/vi4_formats.h
index b1ca0f285..a74f76bf5 100644
--- a/drivers/media/platform/tegra/camera/vi/vi4_formats.h
+++ b/drivers/media/platform/tegra/camera/vi/vi4_formats.h
@@ -120,8 +120,8 @@ static const struct tegra_video_format vi4_video_formats[] = {
 				RAW12, SBGGR12, "BGBG.. GRGR.."),
 
 	/* RGB888 */
-	TEGRA_VIDEO_FORMAT(RGB888, 24, RGB888_1X24, 4, 1, T_A8R8G8B8,
-				RGB888, ABGR32, "BGRA-8-8-8-8"),
+	//TEGRA_VIDEO_FORMAT(RGB888, 24, RGB888_1X24, 4, 1, T_A8R8G8B8,
+	//			RGB888, ABGR32, "BGRA-8-8-8-8"),
 	TEGRA_VIDEO_FORMAT(RGB888, 24, RGB888_1X32_PADHI, 4, 1, T_A8B8G8R8,
 				RGB888, RGB32, "RGB-8-8-8-8"),
 
-- 
2.17.1

